### PR TITLE
added coinbase transaction

### PR DESCRIPTION
**Pull request by flokol120**
* Added coinbase transaction to the data of a block in order to claim a miners reward
* added /balance to get the current balance only based on mining rewards

### CODE REVIEW

File: src/misc/helper.ts [hunk 1]
```diff
@@ -850,7 +909,7 @@ export const verifyBlock = (block: Block): Array<number | boolean> => {
 
 export const verifyReward = (block: Block): number[] => {
   const response = [];
-  let reward = 0;
+  let reward = .0;
```
Comment (BrunoScheufler) [inline src/misc/helper.ts]: What is this format supposed to return? `0.1`? We should document this 👍

File: src/misc/helper.ts [hunk 2]
```diff
@@ -872,10 +931,62 @@ export const verifyReward = (block: Block): number[] => {
   return response;
 };
 
+const getReward = (transactions: Transaction[]): number => {
+  let reward = .0;
```
Comment (BrunoScheufler) [inline src/misc/helper.ts]: This one as well

File: src/misc/helper.ts (hunk 1 above)
Comment (flokol120) [reply src/misc/helper.ts]: .0 = 0.0 Just to document those values can be floating numbers.

Comment (flokol120) [reply src/misc/helper.ts]: .1 would be 0.1

Comment (BrunoScheufler) [reply src/misc/helper.ts]: I think we should add this as a comment as well so nobody gets confused about this 👌

Comment (flokol120) [reply src/misc/helper.ts]: Okay. I guess this is just a strange habit I developed using my CAS-Calculator...

Comment (flokol120) [reply src/misc/helper.ts]: fixed with 968219f

**Comment by flokol120** [comment]
@BrunoScheufler. Our Coinbase is just a specific Transaction which has neither an input- nor an outputValue. The miner is just providing its public key to "receive" its JitCoins. Additionally the miner is signing the signatures of the transactions as well as generating the final random hash by hashing the randomHashes provided by the users.

**Comment by BrunoScheufler** [comment]
@flokol120 because it sounds oddly familiar to an existing platform used to convert currencies to and from Bitcoin 😄 But sure if it's how they call it, I'm fine with that 👍

**Comment by flokol120** [comment]
@BrunoScheufler Ahh you mean Coinbase.com. Well I admit this is basically the same term :smile:.

**Pull request merged**